packaging/os/sorcery.py

```diff
@@ -0,0 +1,637 @@
+
+
+def update_codex(module):
+    """ Update grimoire collections.
+
+    This runs 'scribe update'. Check mode always returns a positive change
+    value when 'cache_valid_time' is used.
+
+    """
+
+    params = module.params
+
+    changed = False
+
+    codex = codex_list(module)
+    fresh = codex_fresh(codex, module)
+
+    if module.check_mode:
+        if not params['name']:
+            if not fresh:
+                changed = True
+
+            module.exit_json(changed=changed, msg="would have updated Codex")
+    elif not fresh or params['name'] and params['state'] == 'latest':
+        # SILENT is required as a workaround for query() in libgpg
+        cmd_scribe = "SILENT=1 %s update" % SORCERY['scribe']
+
+        rc, stdout, stderr = exec_command(cmd_scribe, module)
+
+        if rc != 0:
+            module.fail_json(msg="unable to update Codex: " + stdout)
+
+        if codex != codex_list(module):
+            changed = True
+
+        if not params['name']:
+            module.exit_json(changed=changed,
+                             msg="successfully updated Codex")
+
+
+def match_depends(module):
+    """ Check for matching dependencies.
+
+    This inspects spell's dependencies with the desired states and returns
+    'False' if a recast is needed to match them. It also adds required lines
+    to the system-wide depends file for proper recast procedure.
+
+    """
+
+    params = module.params
+    spells = params['name']
+
+    depends = {}
+
+    depends_ok = True
+
+    if len(spells) > 1 or not params['depends']:
+        return depends_ok
+
+    spell = spells[0]
+
+    if module.check_mode:
+        sorcery_depends_orig = os.path.join(SORCERY_STATE_DIR, "depends")
+        sorcery_depends = os.path.join(SORCERY_STATE_DIR, "depends.check")
+
+        try:
+            shutil.copy2(sorcery_depends_orig, sorcery_depends)
+        except IOError:
+            module.fail_json(msg="failed to copy depends.check file")
+    else:
+        sorcery_depends = os.path.join(SORCERY_STATE_DIR, "depends")
+
+    rex = re.compile(r"^(?P<status>\+?|\-){1}(?P<depend>[a-z0-9]+[a-z0-9_\-\+\.]*(\([A-Z0-9_\-\+\.]+\))*)$")
+
+    for d in params['depends'].split(','):
+        match = rex.match(d)
+
+        if not match:
+            module.fail_json(msg="wrong depends line for spell '%s'" % spell)
+
+        # normalize status
+        if not match.group('status') or match.group('status') == '+':
+            status = 'on'
+        else:
+            status = 'off'
+
+        depends[match.group('depend')] = status
+
+    # drop providers spec
+    depends_list = [s.split('(')[0] for s in depends]
+
+    cmd_gaze = "%s -q version %s" % (SORCERY['gaze'], ' '.join(depends_list))
+
+    rc, stdout, stderr = exec_command(cmd_gaze, module)
+
+    if rc != 0:
+        module.fail_json(msg="wrong dependencies for spell '%s'" % spell)
+
+    fi = fileinput.input(sorcery_depends, inplace=True)
```

Comment:
if the open fails you'll get a fatal error, this should be within an exception handler